File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        int n=numRows;
        List<List<Integer>> res=new ArrayList<>();
        for(int i=0;i<n;i++){
            List<Integer> a=new ArrayList<>();
            for(int j=0;j<=i;j++){
                if(j==0||j==i){
                    a.add(1);
                }
                else{
                    a.add(res.get(i-1).get(j-1)+res.get(i-1).get(j));
                }
            }
            res.add(a);
        }
        return res;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n) {
        vector<vector<int>> a;
        for(int i=0;i<n;i++){
            vector<int> a1;
            for(int j=0;j<=i;j++){
                if(j==0 || j==i) a1.push_back(1);
                else{
                    a1.push_back(a[i-1][j-1]+a[i-1][j]);
                }
            }
            a.push_back(a1);
        }
        return a;
    }
};